### PR TITLE
Divider function test

### PR DESCRIPTION
**Pull request by verasraul**
<h2>Divider Func added</h2>
<h4>created/added unittesting module</h4>

### Test Evidence

```
MACpro:DividerFunc verasraul$ python3 -m unittest dividerFunction1_test.py
......
----------------------------------------------------------------------
Ran 6 tests in 0.001s

OK
```

### CODE REVIEW

**Comment by verasraul** [comment]
@agaro1121

File: PracticeCode/AntProjects/dividerFunction1.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#Simple function takes two integers as arguments and divides the first int by the second int.
+#Result will include remainders if any.
+
+
+def divider (number1, number2):
+    # Divide number1 by number2
```
Comment (agaro1121) [inline PracticeCode/AntProjects/dividerFunction1.py]: Make this a doc above the function name with the following information:
- param name
- accepted type - Integer/Float
- description
- return type

File: PracticeCode/AntProjects/dividerFunction1.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#Simple function takes two integers as arguments and divides the first int by the second int.
+#Result will include remainders if any.
+
+
+def divider (number1, number2):
+    # Divide number1 by number2
+    division = number1 / number2
+    return (division)
```
Comment (agaro1121) [inline PracticeCode/AntProjects/dividerFunction1.py]: you can simply write: `return (number1/number2)`

File: PracticeCode/AntProjects/dividerFunction1.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+
+def divider (number1, number2):
+    # Divide number1 by number2
+    division = number1 / number2
+    return (division)
+
+
+#divider(3, 0)
+#divider(3, 1)
```
Comment (agaro1121) [inline PracticeCode/AntProjects/dividerFunction1.py]: Delete commented code

File: PracticeCode/AntProjects/dividerFunction1_test.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import unittest
+from dividerFunction1 import divider
+
+# Test for dividerFunction1 pycode.
+# Test 1; func will throw an error when dividing by zero
+# Test 2; func will return a float when dividing by an Int by a float
+
+
+#divider(3,1)
```
Comment (agaro1121) [inline PracticeCode/AntProjects/dividerFunction1_test.py]: Delete commented code

File: PracticeCode/AntProjects/dividerFunction1.py (hunk 1 above)
Comment (agaro1121) [reply PracticeCode/AntProjects/dividerFunction1.py]: See code from python/python for examples

**Review by agaro1121 — CHANGES_REQUESTED** [review]
The binary files look like they don't need to be there. Please add a gitignore file
- does this project follow proper folder structure? Normally, test code isn't bundled with production code

File: PracticeCode/AntProjects/dividerFunction1_test.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        self.assertEqual(result, expected_result)
+
+
+
+
+
+
+
+
+if __name__=="__main__":
```
Comment (agaro1121) [inline PracticeCode/AntProjects/dividerFunction1_test.py]: This doesn't look right. Is this the proper way to run unit tests?

File: PracticeCode/AntProjects/dividerFunction1_test.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        self.assertRaises(ZeroDivisionError, divider, 3, 0)
+
+    def test_float_result(self):
+        # capture result of divider func
+        result = divider(3, 1.0)
+        # expected result of func
+        expected_result = 3.0
+        self.assertEqual(result, expected_result)
+
+
```
Comment (agaro1121) [inline PracticeCode/AntProjects/dividerFunction1_test.py]: Add more tests.
- what happens when number2 > number1?
- what happens when number1 == number2?
- what happens when they're both floats?
- what happens when they're both Ints?
You don't need to check the type. You can simply check the return type like above

**Review by agaro1121 — CHANGES_REQUESTED** [review]

File: PracticeCode/AntProjects/dividerFunction1_test.py (hunk 6 above)
Comment (verasraul) [reply PracticeCode/AntProjects/dividerFunction1_test.py]: Added new funcs to match request above.

File: PracticeCode/AntProjects/dividerFunction1_test.py (hunk 5 above)
Comment (verasraul) [reply PracticeCode/AntProjects/dividerFunction1_test.py]: Well you can call it from the terminal with ```python3 -m unittest filename``` but in order to run it by executing the file I believe I had to add those lines.

**Review by verasraul — COMMENTED** [review]
opened review by mistake just testing around.

File: PracticeCode/AntProjects/dividerFunction1.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
+#Simple function takes two integers as arguments and divides the first int by the second int.
+#Result will include remainders if any.
+
+
+def divider (number1, number2):
+    # Func takes 2 ints as arguments.
+    # Return the result of number1 divided by number2.
```
Comment (agaro1121) [inline PracticeCode/AntProjects/dividerFunction1.py]: update comment like [this](https://github.com/python/cpython/blob/master/Lib/email/header.py#L61-L74)

File: PracticeCode/AntProjects/dividerFunction1_test.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    def test_division_by_higher_number(self):
+        # Test func when both numbers divided are the same
+        self.assertEqual(divider(3, 6), 0.5)
+
+
+
+
+
+
+if __name__=="__main__":
```
Comment (agaro1121) [inline PracticeCode/AntProjects/dividerFunction1_test.py]: delete this

Comment (agaro1121) [reply PracticeCode/AntProjects/dividerFunction1_test.py]: or you need this?

**Comment by agaro1121** [comment]
add `.gitignore` file

File: PracticeCode/AntProjects/dividerFunction1.py [hunk 9]
```diff
@@ -0,0 +1,10 @@
+#Simple function takes two integers as arguments and divides the first int by the second int.
+#Result will include remainders if any.
```
Comment (agaro1121) [inline PracticeCode/AntProjects/dividerFunction1.py]: delete this

**Comment by agaro1121** [comment]
Please add test evidence (terminal output from unit tests) to the description under :
### Test Evidence

```
MACpro:DividerFunc verasraul$ python3 -m unittest dividerFunction1_test.py
......
----------------------------------------------------------------------
Ran 6 tests in 0.001s

OK
```

**Comment by agaro1121** [comment]
Use [this](https://github.com/python/cpython/blob/master/Lib/email/header.py#L61-L74) as an example of how to organize tests and run them

File: PracticeCode/AntProjects/dividerFunction1_test.py (hunk 8 above)
Comment (verasraul) [reply PracticeCode/AntProjects/dividerFunction1_test.py]: According to the documentation it is the it's used int he implementation.

**Review by agaro1121 — APPROVED** [review]

**Pull request merged**